src/coord/timestamp.rs

```diff
@@ -310,6 +312,30 @@ struct ByoKafkaConnector {
     last_watermark_update: Instant,
 }
 
+// A 0-index offset wrapper for Kafka connectors
```

Comment:
Can you add something explaining that KafkaOffsets represent a message, whereas MzOffsets represent "the set of messages whose offset is < a given kafka offset" (or something to that effect). I think just saying 0-indexed vs. 1-indexed is a bit too under-motivated / possibly confusing.

Reply:
I don't think that's right? MzOffset at x does just represent a message at KafkaOffset x-1

Reply:
I was thinking of it as being “the address of the next offset” rather than “one less than the address of the current offset” (those end up meaning the same thing, of course)

Reply:
Sorry, I'm not sure I follow, would you mind expanding?

Reply:
The MzOffset corresponding to an empty stream is 0, i.e., the _next_ Kafka offset.

Reply:
I'm happy to write that if you want and it makes it clearer for you, but I find it a bit confusing. We don't use MzOffsets to represent the "next kafka offset", but we use them to talk about *current* offsets.

Reply:
but there might not be any current offset (if the stream is empty), in which case we use zero, right?

Reply:
It's hard for me to interpret zero there as anything other than "the next available Kafka offset"

Reply:
Yes that's correct, 0 denotes the absence of a message.

Reply:
so an MzOffset of 0 means something categorically different from 1, 2, ... ?

In that case, why not just use an option?